### PR TITLE
Update Travis to JSLINT all changes

### LINKED ISSUE #414: Add Lint checking to Travis
Travis should run a jsonlint check on all files modified to ensure they do not break formatting.

### PR DESCRIPTION
**Pull request by ZeldaZach**
Fixes #414

This script should
* Install the jslint NPM module
* Run a git check to see what files were edited in this PR
* For each JSON file, run `jslint` on it
    * If any file fails, return out and fail travis

### CODE REVIEW

File: util/travis.sh [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
+
+	# Make sure the changelog was updated
     if git diff --quiet master HEAD -- web/changelog.json; then
         echo 'All pull requests must update the changelog.'
         exit 1
     fi
-fi
+
+	# Make sure all changed JSON files are jslint compliant
+	npm install jslint -g
```
Comment (gwax) [inline util/travis.sh]: Please do not install packages into other people's global namespace.

File: util/travis.sh [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
+	git diff --name-only master HEAD | grep .json | while read changedFile; do
```
Comment (tritoch) [inline util/travis.sh]: This should only look in `./json/` directory, package.json won't pass lint, and maybe others.

Comment (ZeldaZach) [inline util/travis.sh]: Any reason it wouldn't pass lint? And I want to check changelog and such.

File: util/travis.sh (hunk 1 above)
Comment (ZeldaZach) [inline util/travis.sh]: How should this be done then? I need a way to call this via bash.

File: util/travis.sh (hunk 2 above)
Comment (tritoch) [inline util/travis.sh]: Nevermind, thought this was running against the schemas.

Comment (ZeldaZach) [inline util/travis.sh]: You know, I didn't think about it. I'm validating it's proper JSON but didn't check if the sets are valid in their own rights... a thought to change

File: .travis.yml [hunk 3]
```diff
@@ -7,6 +7,8 @@ node_js:
 
 script:
   - npm test
+  - npm checkJSON
```
Comment (gwax) [inline .travis.yml]: this needs to be `npm run checkJSON` (note the **run**)

File: util/travis.sh [hunk 4]
```diff
@@ -1,9 +1,10 @@
 #!/bin/bash
 
-# make sure pull requests include a changelog
```
Comment (gwax) [inline util/travis.sh]: Since the comment changes are unrelated to your other PR changes, could you please revert this file to master?

File: package.json [hunk 5]
```diff
@@ -21,12 +21,16 @@
...
+    "jsonschema-cli": "^0.1.2"
   },
   "engines": {
     "node": ">= 0.12.0"
   },
   "scripts": {
-    "test": "./node_modules/jshint/bin/jshint build/*.js shared/*.js util/*.js"
+    "test": "jshint build/*.js shared/*.js util/*.js",
+    "checkJSON": "jslint --edition=latest cron/*.json json/*.json decks/*.json shared/set_configs/**/*.json util/**/*.json web/*.json > /dev/null",
+    "validateSchema": "validateJson -s json/schema/singleSet.schema.json -d json/*.json > /dev/null"
```
Comment (ZeldaZach) [inline package.json]: NOTE: Maybe try out https://www.npmjs.com/package/jsonlint-cli

**Comment by ZeldaZach** [comment]
Inspiration from https://stackoverflow.com/questions/884650/how-to-spawn-parallel-child-processes-on-a-multi-processor-system

**Comment by ZeldaZach** [comment]
Closed in favor of #431
